### DISCUSSION TITLE
TCGPlayer Please

### DISCUSSION THREAD

**Discussion by SimpleSon** [StepKie/MtgCsvHelper#9] [General]
I just came across this after reading several threads on collection managers.  I've already scanned my collection into TCGP, but I want to move everything to manabox.  The default CSV export / import between the two only found about half of the cards in that collection.

**Comment by StepKie** [comment]
can you please share the csv that does not convert correctly?
And if you can tell me what exactly did not work as expected, that would be helpful. Thanks.

**Reply by StepKie** [reply]
@SimpleSon hi, did you see my followup question? Otherwise I can not really get started on this task.

**Reply by SimpleSon** [reply]
Sorry for missing this reply, and thank you for reaching out.  

Just at first pass the import was only able to detect over 2000 cards, but the collection is over 4000 cards.  I didn't actually dive much deeper into the imports to confirm further. 

[new cards (1).csv](https://github.com/user-attachments/files/16896803/new.cards.1.csv)
[old cards (1).csv](https://github.com/user-attachments/files/16896804/old.cards.1.csv)

**Reply by StepKie** [reply]
@SimpleSon ok, please try with the newest version and let me know if this resolves your problem

I could not find any problems when importing your csv's ...
But I have added TcgPlayer to the selectable formats in the web tool as well, and fixed a problem of not being able to convert due to an external Scryfall API change.

**Comment by SimpleSon** [comment]
Thank you for the update!

 I ran both my vintage and modern collections through the web app, and I
had 50 and 38 errors respectively. Unfortunately, I didn't open the error
log before saving the import, and I realize now that I can't easily find
that log after completing the import. I've noticed at least 1 card that is
missing, but I'm sure there are a few more.

Overall, for whatever reason, I got significantly better results on this
pass, and I'll assume last time was a fluke since you had similar results.

Thank you for the updates!  Everything worked quite well otherwise and I
really enjoy having migrated the majority of my collections to Manabox over
TCGPlayer.  I'll need to find what I'm missing eventually, but hopefully
it's < 100 cards.

- Reed

On Sun, Sep 8, 2024 at 10:51 AM Stephan Kieburg ***@***.***>
wrote:

> @SimpleSon <https://github.com/SimpleSon> ok, please try with the newest
> version and let me know if this resolves your problem
>
> I could not find any problems when importing your csv's ...
> But I have added TcgPlayer to the selectable formats in the web tool as
> well, and fixed a problem of not being able to convert due to an external
> Scryfall API change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StepKie/MtgCsvHelper/discussions/9#discussioncomment-10583375>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANR6SY6TRG7W454EE5FADJTZVR6AJAVCNFSM6AAAAABNJWSOK6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJYGMZTONI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Reed Simpson*
e: ***@***.***
t: (720) 443-0727

**Reply by StepKie** [reply]
What errors, after importing the resulting csv into manabox, or in the webapp? where was that displayed? _How can I recreate your errors? Can you share the exact csvs and/or logs_?
I really dislike leaving a state like "oh well, there are still lots of errors, but it is better than before" ^^

I cant really see the issue during the conversion, I tested with both csvs you provided and it converted all the cards.
You mean the "log" from the import in the manabox app?

----

I think one change I made that might have alleviated some problems was reading the card name from the "Simple Name" instead of the "Name" column, since TcgPlayer writes something weird in the "Name" column for certain printings, like

Quantity,Name,Simple Name,Set,Card Number,Set Code,Printing,Condition,Language,Rarity,Product ID,SKU
2,Ertai (Oversize),Ertai,Vanguard,101,VNG,Normal,Near Mint,English,Promo,21607,129896

and then other collection managers have trouble importing from that name.

**Reply by StepKie** [reply]
@SimpleSon see above, tagging you one last time for visibility

**Comment by SimpleSon** [comment]
Sorry for the late reply.  I just ran the imports again and was able to
download the lines where manabox had an error and didn't import.

**Reply by StepKie** [reply]
> Sorry for the late reply. I just ran the imports again and was able to download the lines where manabox had an error and didn't import.
> […](#)

would you mind sharing them?

**Reply by SimpleSon** [reply]
Here is the output from Manabox's error logs.  

[lines_errors.csv](https://github.com/user-attachments/files/17004708/lines_errors.csv)
[new_cards_lines_errors.csv](https://github.com/user-attachments/files/17004711/new_cards_lines_errors.csv)

**Reply by StepKie** [reply]
HI @SimpleSon,

I have looked at the errors:

as you have probably noticed yourself, they all relate to cards that are either from a promo sheet, or have a highly irregular format and/or name:

Multani,1,303,VNG,Vanguard,near_mint,normal,en,,"Line 1047 incorrect: Set not found ""VNG"""
Warren Warleader,1,38,PPBLB,Promo Pack: Bloomburrow,near_mint,normal,en,,"Line 784 incorrect: Set not found ""PPBLB"""

They use their own set names which are incompatible with the official set names (oracle, scryfall), and sometimes even the card or set names do not even match any official syntax. For example, neither a card with the name "Multani", nor a set with the name "Vanguard" exists.

Thus, it is hard to make a conversion (i.e. "fix" the set codes, names, etc.) when there is no documentation available from TcgPlayer or other sites what these names are.

So long story short, I am very sorry but I don't see this happening anytime soon since I don't see a clear way forward to implement this feature that would scale. i.e. I could "fix" the occurrences that you have reported by doing manual conversions, or adding them to a configurable mapping list, but this approach seems very hard to support across a larger number of csv formats/vendors (>10).

Best regards
Stephan

**Comment by SimpleSon** [comment]
Thanks for the feedback. Feel free to close this out.  I should be able to
take care of the manual entries.  Those old vanguard cards are an oddity
today.

On Sun, Sep 29, 2024 at 6:26 AM Stephan Kieburg ***@***.***>
wrote:

> HI @SimpleSon <https://github.com/SimpleSon>,
>
> I have looked at the errors:
>
> as you have probably noticed yourself, they all relate to cards that are
> either from a promo sheet, or have a highly irregular format and/or name:
>
> Multani,1,303,VNG,Vanguard,near_mint,normal,en,,"Line 1047 incorrect: Set
> not found ""VNG"""
> Warren Warleader,1,38,PPBLB,Promo Pack:
> Bloomburrow,near_mint,normal,en,,"Line 784 incorrect: Set not found
> ""PPBLB"""
>
> They use their own set names which are incompatible with the official set
> names (oracle, scryfall), and sometimes even the card or set names do not
> even match any official syntax. For example, neither a card with the name
> "Multani", nor a set with the name "Vanguard" exists.
>
> Thus, it is hard to make a conversion (i.e. "fix" the set codes, names,
> etc.) when there is no documentation available from TcgPlayer or other
> sites what these names are.
>
> So long story short, I am very sorry but I don't see this happening
> anytime soon since I don't see a clear way forward to implement this
> feature that would scale. i.e. I could "fix" the occurrences that you have
> reported by doing manual conversions, or adding them to a configurable
> mapping list, but this approach seems very hard to support across a larger
> number of csv formats/vendors (>10).
>
> Best regards
> Stephan
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StepKie/MtgCsvHelper/discussions/9#discussioncomment-10788803>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANR6SY4D4PYWICKZCHW4YZ3ZY7WYXAVCNFSM6AAAAABNJWSOK6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZYHA4DAMY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Reed Simpson*
e: ***@***.***
t: (720) 443-0727
